### PR TITLE
Update to use #if SNIPPET in core and appconfig

### PR DESCRIPTION
**Pull request by pakrym**
Contributes to https://github.com/Azure/azure-sdk-for-net/issues/20431

### CODE REVIEW

File: sdk/core/Azure.Core/tests/AzureNamedKeyCredentialTests.cs [hunk 1]
```diff
@@ -73,9 +73,10 @@ public void UpdateCanBePerformed()
 
             #region Snippet:AzureNamedKeyCredential_Deconstruct
             var credential = new AzureNamedKeyCredential("SomeName", "SomeKey");
-            /*@@*/
-            /*@@*/ credential.Update(expectedName, expectedKey);
-            /*@@*/
+
+            #if !SNIPPET
```
Comment (heaths) [inline sdk/core/Azure.Core/tests/AzureNamedKeyCredentialTests.cs:77]: Nit: You might get some complaints from Roslyn about not being at char 0.

File: sdk/core/Microsoft.Azure.Core.NewtonsoftJson/tests/Samples/Readme.cs [hunk 2]
```diff
@@ -49,7 +49,9 @@ public void SearchSample()
 
             SearchClientOptions clientOptions = new SearchClientOptions
             {
+#if !SNIPPET
                 /*@@*/ Transport = new MockTransport(response),
```
Comment (heaths) [inline sdk/core/Microsoft.Azure.Core.NewtonsoftJson/tests/Samples/Readme.cs:53]: Forgot to remove the comment chars.

File: sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs [hunk 3]
```diff
@@ -60,7 +60,9 @@ public void SearchSample()
 
             SearchClientOptions clientOptions = new SearchClientOptions
             {
+#if !SNIPPET
                 /*@@*/ Transport = new MockTransport(response),
```
Comment (heaths) [inline sdk/core/Microsoft.Azure.Core.Spatial.NewtonsoftJson/tests/Samples/Readme.cs:64]: Forgot to remove the comment chars.

File: sdk/core/Microsoft.Azure.Core.Spatial/tests/Samples/Readme.cs [hunk 4]
```diff
@@ -59,7 +59,9 @@ public void SearchSample()
 
             SearchClientOptions clientOptions = new SearchClientOptions
             {
+#if !SNIPPET
                 /*@@*/ Transport = new MockTransport(response),
```
Comment (heaths) [inline sdk/core/Microsoft.Azure.Core.Spatial/tests/Samples/Readme.cs:63]: Forgot to remove the comment chars.

**Review by heaths — APPROVED** [review]
Just a few minor comments but otherwise LGTM.

File: sdk/core/Azure.Core/src/AzureNamedKeyCredential.cs [hunk 5]
```diff
@@ -64,7 +64,8 @@ public void Update(string name, string key)
         /// <example>
         /// <code snippet="Snippet:AzureNamedKeyCredential_Deconstruct">
         /// var credential = new AzureNamedKeyCredential(&quot;SomeName&quot;, &quot;SomeKey&quot;);
-        /// (string name, string key) = credential;
+        ///
+        ///                         (string name, string key) = credential;
```
Comment (maririos) [inline sdk/core/Azure.Core/src/AzureNamedKeyCredential.cs:68]: nit: not sure these extra spaces were intentional...

**Review by maririos — APPROVED** [review]

**Pull request merged**